### PR TITLE
Quote primary table name if identifier quoting is enabled

### PR DESCRIPTION
**Pull request by nederdirk**
With the latest `master` build we ran into the problem that `COUNT(*)`-queries to our `order` table failed. This patch ensures that the setting for identifier quoting is properly observed.

Note that I deliberately ignored the `getSql` helper-function in the test, as this behaviour is precisely what the `getSql` helper-function hides.

### CODE REVIEW

**Review by dereuromark — APPROVED** [review]
Was this a regression recently introduced?

**Comment by nederdirk** [comment]
> Was this a regression recently introduced?

I think so, as we didn't encounter the same problem before. However, I'm not entirely sure and I'm not interested in the finger-pointing game :smile_cat:

**Comment by dereuromark** [comment]
All good - if anything we would rather use this to improve our code review skills :) After all, we approved and merged without seeing this.
Anyway, we do need a 2nd signoff on this to merge.

**Review by yaroslav-spryker — APPROVED** [review]

**Pull request merged**
